formats/opentype.fathom

```diff
@@ -224,33 +302,228 @@ let byte_encoding_table = fun (platform : Repr platform_id) => {
+    /// Set to 0.
+    _reserved_pad <- reserved s16be 0,
+
+    // TODO these need arithmetic
+    // /// Start character code for each segment.
+    // start_code[segCount]  <- u16be,
+    // /// Delta for all character codes in segment.
+    // id_delta[segCount]  <- int16,
+    // /// Offsets into glyphIdArray or 0
+    // id_range_offsets[segCount]  <- u16be,
+    // /// Glyph index array (arbitrary length)
+    // glyph_id_array[ ]  <- u16be,
+};
+
+/// Format 6: Trimmed table mapping
+///
+/// Format 6 was designed to map 16-bit characters to glyph indexes when the character codes for a
+/// font fall into a single contiguous range.
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Format 6: Segment mapping to delta values](https://docs.microsoft.com/en-us/typography/opentype/spec/cmap#format-6-trimmed-table-mapping)
+/// - [Apple's TrueType Reference Manual: `'cmap'` format 6](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6cmap.html)
+let cmap_subtable_format6 = fun (platform : Repr platform_id) => {
+    /// The length of the subtable in bytes
+    length <- u16be,
+    /// The language ID of the subtable
+    language <- cmap_language_id platform,
+    /// First character code of subrange.
+    first_code <- u16be,
+    /// Number of character codes in subrange.
+    entry_count <- u16be,
+    /// Array of glyph index values for character codes in the range.
+    glyph_id_array <- array16 entry_count u16be,
+};
+
+/// Format 8: mixed 16-bit and 32-bit coverage
+///
+/// Subtable format 8 was designed to support Unicode supplementary-plane characters in UTF-16
+/// encoding, though it is not commonly used. Format 8 is similar to format 2, in that it provides
+/// for mixed-length character codes. Instead of allowing for 8- and 16-bit character codes,
+/// however, it allows for 16- and 32-bit character codes.
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Format 8: Segment mapping to delta values](https://docs.microsoft.com/en-us/typography/opentype/spec/cmap#format-8-mixed-16-bit-and-32-bit-coverage)
+/// - [Apple's TrueType Reference Manual: `'cmap'` format 8](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6cmap.html)
+let cmap_subtable_format8 = fun (platform : Repr platform_id) => {
+    /// Set to 0.
+    _reserved <- reserved u16be 0,
+    /// The length of the subtable in bytes (including the header)
+    length <- u32be,
+    /// The language ID of the subtable
+    language <- cmap_language_id32 platform,
+    /// Tightly packed array of bits (8K bytes total) indicating whether the particular 16-bit
+    /// (index) value is the start of a 32-bit character code
+    is32 <- array16 8192 u8,
+    /// Number of groupings which follow
+    num_groups <- u32be,
+    /// Array of SequentialMapGroup records.
+    groups <- array32 num_groups sequential_map_group,
+};
+
+/// Format 10: Trimmed table mapping
```

Comment:
Some of  these doc comments use inconsistent headings - not sure if it really matter at this stage though!

Reply:
Ahh, there's no way to extract the doc comments yet is there?

Reply:
No :(

Was hoping one day to be able to extract a specification site. And also with the binary grep tooling you could get explanations of fields/types somehow. But yeah that is definitely a way off.

Reply:
I fixed the Markdown

Reply:
Looks good! 👍